source/common/upstream/load_balancer_impl.cc

```diff
@@ -727,7 +727,10 @@ void EdfLoadBalancerBase::refresh(uint32_t priority) {
       return;
     }
 
-    scheduler.edf_ = std::make_unique<EdfScheduler<const Host>>();
+    // Current time is initialized as a float ranges [0.0, 1.0).
+    // Accuracy 1e-5 is enough because (1 / 128) ^ 2 > 1e-5. 128 is the largest weight for hosts
```

Comment:
I might be missing something, but why exactly is squaring  the inverse weight necessary in this calculation?

Reply:
I was wrong in this point and `(1 / 128) ^ 2` really makes no sense.
`1e-6` would be better than `1e-5` for probability distributions.
What's should I do if I just want a random number among [0.0, 1.0)?

---

Original thought:
`1/127 - 1/128` is the minimum difference for two hosts' deadline, and
`1/127 - 1/128 ≈ (1/128)^2`. So, 1e-5 could separate different deadline.